pallets/handles/src/lib.rs

```diff
@@ -566,6 +566,13 @@ pub mod pallet {
 
 			let base_handle_str = Self::validate_base_handle(payload.base_handle)?;
 
+			// Strip diacritical marks from the validated handle
+			let diacritical_stripped_base_handle_str = strip_diacriticals(&base_handle_str);
+
+			Self::validate_base_handle_contains_characters_in_supported_ranges(
+				&diacritical_stripped_base_handle_str,
+			)?;
+
```

Comment:
Our diacritical stripping function removes diacritical marks from UTF8 encoded strings where both a base character plus combining mark AND a single character containing both base and mark are present. Stripping diacriticals before doing the validation of supported ranges allows: 

1. a handle to be submitted with either combination of encoded forms
2. a reduction of the size of the `CONFUSABLES` map by eliminating the need to add characters with diacriticals from the map, thereby reducing the final WASM size